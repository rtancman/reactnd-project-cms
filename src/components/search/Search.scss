.search__bar {
  &__result{
    width: auto;
    margin-left: 8px;

    * {
      color: #000;
    }

    .search__bar__result__box{
      border-radius: 0 0 4px 4px;
    }
  }
}