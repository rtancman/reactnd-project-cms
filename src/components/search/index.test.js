import React from 'react'
import { shallow, mount } from 'enzyme'
import Search from './'
import { MemoryRouter } from 'react-router';
import Input from '@material-ui/core/Input';

describe('Search Component', () => {
  const history = jest.fn()
  const options = [
    { name: 'article 1', path: '/article/1' },
    { name: 'post 1', path: '/post/2' },
    { name: 'article 2', path: '/article/2' },
    { name: 'post 2', path: '/post/2' },
  ]

  describe('when component renders', () => {
    it('without crashing', () => {
      const props = {
        options: [],
        history,
      }
      const wrapper = shallow(<Search {...props} />)

      expect(wrapper).toMatchSnapshot()
    })
  })

  describe('when search have result', () => {
    describe('when component renders', () => {
      it('display children components', () => {
        const props = {
          options,
        }
        const root = mount(
          <MemoryRouter>
            <Search {...props} />
          </MemoryRouter>
        )
        
        const search = root.find('Search')
        expect(search.length).toBe(1)
        expect(search.find('Input').length).toBe(1)
        expect(search.find('SearchIcon').length).toBe(1)
        expect(search.find('Paper').length).toBe(1)
        expect(search.find('Paper').text()).toEqual('')
      })
    })

    describe('when user put word in search', () => {

      it('simulate change event for input', () => {
        const props = {
          options,
        }
        const root = mount(
          <MemoryRouter>
            <Search {...props} />
          </MemoryRouter>
        )

        const search = root.find('Search')
        search.instance().onSearchDebounced = search.instance().onSearch //@TODO change this hack ;)
        const inputSearch = search.find('Input[name="query"]').find('input')
        inputSearch.instance().value = "article"
        inputSearch.simulate('change');
        expect(search.instance().state.query).toEqual('article')
        expect(search.instance().state.items.length).toBe(2)
      })
    })
  })
})