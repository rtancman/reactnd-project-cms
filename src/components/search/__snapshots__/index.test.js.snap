// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search Component when component renders without crashing 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <withRouter(WithStyles(Search))
    history={[Function]}
    options={Array []}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": [Function],
    },
    "ref": null,
    "rendered": [Function],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": [Function],
      },
      "ref": null,
      "rendered": [Function],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
